### ISSUE TITLE
Q: should trigger_error type E_USER_DEPRECATED cause testing to halt?

### ISSUE THREAD

**Issue by adamcameron** [kahlan/kahlan#375]
I have discovered this code in Symfony, thanks to Kahlan barfing at it:

```
trigger_deprecation('', '', 'Please install the "intl" PHP extension for best performance.');
```

Which in turns calls `@trigger_error` with `E_USER_DEPRECATED`. See:

```
vendor/symfony/deprecation-contracts/function.php, line  to 25
vendor/symfony/framework-bundle/DependencyInjection/FrameworkExtension.php, line  to 253
```

To me, given the `trigger_error` call is `@`-ed, and the error is a *deprecation*, then intrinsically processing should continue.

However Kahlan decides to halt with:

```
      an uncaught exception has been thrown in `vendor/symfony/deprecation-contracts/function.php` line 25

      message:`Kahlan\PhpErrorException` Code(0) with message "`E_USER_DEPRECATED` Please install the \"intl\" PHP extension for best performance."
```

Yeah, fair cop I should have the intl extension loaded, but... I think Kahlan is probably overstepping here in deciding a silenced deprecation notice is something to halt at.

**Comment by jails** [comment]
Fixed in the 5.1.2 release

**Comment by eexit** [comment]
Hello,

Sorry to dig this out but I'm facing the same issue. What's weird is that I can't reproduce locally, only in my CI.

```
App\Symfony\Form\Type\UserPasswordType
    createNew
      ✖ it submit under 12 characters generate error
        an uncaught exception has been thrown in `vendor/symfony/translation/Util/XliffUtils.php` line 65
        
        message:`Kahlan\PhpErrorException` Code(0) with message "`E_DEPRECATED` Function libxml_disable_entity_loader() is deprecated"
        
          [NA] - vendor/symfony/translation/Util/XliffUtils.php, line  to 65
          Symfony\Component\Translation\Util\XliffUtils::validateSchema() - vendor/symfony/translation/Loader/XliffFileLoader.php, line 61
          Symfony\Component\Translation\Loader\XliffFileLoader::load() - vendor/symfony/translation/Translator.php, line 364
          Symfony\Component\Translation\Translator::doLoadCatalogue() - vendor/symfony/framework-bundle/Translation/Translator.php, line 140
          Symfony\Bundle\FrameworkBundle\Translation\Translator::doLoadCatalogue() - vendor/symfony/translation/Translator.php, line 256
          Symfony\Component\Translation\Translator::initializeCatalogue() - vendor/symfony/framework-bundle/Translation/Translator.php, line 132
          Symfony\Bundle\FrameworkBundle\Translation\Translator::initializeCatalogue() - vendor/symfony/translation/Translator.php, line 375
          Symfony\Component\Translation\Translator::loadFallbackCatalogues() - vendor/symfony/translation/Translator.php, line 262
          Symfony\Component\Translation\Translator::initializeCatalogue() - vendor/symfony/framework-bundle/Translation/Translator.php, line 132
          Symfony\Bundle\FrameworkBundle\Translation\Translator::initializeCatalogue() - vendor/symfony/translation/Translator.php, line 290
          Symfony\Component\Translation\Translator::dumpCatalogue() - vendor/symfony/translation/Translator.php, line 275
          Symfony\Component\Config\ResourceCheckerConfigCacheFactory::cache() - vendor/symfony/translation/Translator.php, line 273
          Symfony\Component\Translation\Translator::initializeCacheCatalogue() - vendor/symfony/translation/Translator.php, line 244
          Symfony\Component\Translation\Translator::loadCatalogue() - vendor/symfony/translation/Translator.php, line 215
          Symfony\Component\Translation\Translator::getCatalogue() - vendor/symfony/translation/Translator.php, line 182
          Symfony\Component\Translation\Translator::trans() - vendor/symfony/translation/DataCollectorTranslator.php, line 45
          Symfony\Component\Translation\DataCollectorTranslator::trans() - src/Symfony/Translation/Translator.php, line 43
          App\Symfony\Translation\Translator::trans() - vendor/symfony/validator/Violation/ConstraintViolationBuilder.php, line 127
          Symfony\Component\Validator\Violation\ConstraintViolationBuilder::addViolation() - vendor/symfony/validator/Constraints/LengthValidator.php, line 99
          Symfony\Component\Validator\Constraints\LengthValidator::validate() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 752
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateInGroup() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 597
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateGenericNode() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 106
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validate() - vendor/symfony/form/Extension/Validator/Constraints/FormValidator.php, line 138
          Symfony\Component\Form\Extension\Validator\Constraints\FormValidator::validate() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 752
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateInGroup() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 597
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateGenericNode() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 106
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validate() - vendor/symfony/form/Extension/Validator/Constraints/FormValidator.php, line 145
          Symfony\Component\Form\Extension\Validator\Constraints\FormValidator::validate() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 752
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateInGroup() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 485
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateClassNode() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 301
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validateObject() - vendor/symfony/validator/Validator/RecursiveContextualValidator.php, line 131
          Symfony\Component\Validator\Validator\RecursiveContextualValidator::validate() - vendor/symfony/validator/Validator/RecursiveValidator.php, line 87
          Symfony\Component\Validator\Validator\RecursiveValidator::validate() - vendor/symfony/validator/Validator/TraceableValidator.php, line 61
          Symfony\Component\Validator\Validator\TraceableValidator::validate() - vendor/symfony/form/Extension/Validator/EventListener/ValidationListener.php, line 49
          Symfony\Component\Form\Extension\Validator\EventListener\ValidationListener::validateForm() - vendor/symfony/event-dispatcher/EventDispatcher.php, line 220
          Symfony\Component\EventDispatcher\EventDispatcher::callListeners() - vendor/symfony/event-dispatcher/EventDispatcher.php, line 56
          Symfony\Component\EventDispatcher\EventDispatcher::dispatch() - vendor/symfony/event-dispatcher/ImmutableEventDispatcher.php, line 30
          Symfony\Component\EventDispatcher\ImmutableEventDispatcher::dispatch() - vendor/symfony/form/Form.php, line 591
          Symfony\Component\Form\Form::submit() - tests/integration/Symfony/Form/Type/UserPasswordType.spec.php, line 32
          Kahlan\Filter\Filters::run() - vendor/kahlan/kahlan/src/Suite.php, line 239
          Kahlan\Suite::runBlock() - vendor/kahlan/kahlan/src/Block/Specification.php, line 112
          Kahlan\Block\Specification::_execute() - vendor/kahlan/kahlan/src/Block.php, line 380
          Kahlan\Block::_process() - vendor/kahlan/kahlan/src/Block.php, line 208
          Kahlan\Block::process() - vendor/kahlan/kahlan/src/Block/Group.php, line 338
          Kahlan\Block\Group::_execute() - vendor/kahlan/kahlan/src/Block.php, line 380
          Kahlan\Block::_process() - vendor/kahlan/kahlan/src/Block.php, line 208
          Kahlan\Block::process() - vendor/kahlan/kahlan/src/Block/Group.php, line 338
          Kahlan\Block\Group::_execute() - vendor/kahlan/kahlan/src/Block.php, line 380
          Kahlan\Block::_process() - vendor/kahlan/kahlan/src/Block.php, line 208
          Kahlan\Block::process() - vendor/kahlan/kahlan/src/Block/Group.php, line 338
          Kahlan\Block\Group::_execute() - vendor/kahlan/kahlan/src/Block.php, line 380
          Kahlan\Block::_process() - vendor/kahlan/kahlan/src/Block.php, line 208
          Kahlan\Block::process() - vendor/kahlan/kahlan/src/Suite.php, line 218
          Kahlan\Suite::run() - vendor/kahlan/kahlan/src/Cli/Kahlan.php, line 573
          Kahlan\Filter\Filters::run() - vendor/kahlan/kahlan/src/Cli/Kahlan.php, line 573
          Kahlan\Cli\Kahlan::_run() - vendor/kahlan/kahlan/src/Cli/Kahlan.php, line 378
          Kahlan\Filter\Filters::run() - vendor/kahlan/kahlan/src/Cli/Kahlan.php, line 367
          Kahlan\Cli\Kahlan::run() - vendor/kahlan/kahlan/bin/kahlan, line 71
          include - vendor/bin/kahlan, line  to 119
        
      ✓ it submit above 12 characters is valid
```

PHP version:

```
PHP 8.3.27 (fpm-fcgi) (built: Oct 24 2025 19:29:51)
Copyright (c) The PHP Group
Zend Engine v4.3.27, Copyright (c) Zend Technologies
    with Zend OPcache v8.3.27, Copyright (c), by Zend Technologies
    with Xdebug v3.3.1, Copyright (c) 2002-2023, by Derick Rethans
```

Kahlan version:

```
  /\ /\__ _| |__ | | __ _ _ __
 / //_/ _` | '_ \| |/ _` | '_ \
/ __ \ (_| | | | | | (_| | | | |
\/  \/\__,_|_| |_|_|\__,_|_| |_|
The PHP Test Framework for Freedom, Truth and Justice.
version 6.0.2
For additional help you must use --help
```

Here's the offending line: https://github.com/symfony/translation/blob/v6.4.26/Util/XliffUtils.php#L65
